app/frontend/src/features/messages/groupchats/CreateGroupChat.tsx

```diff
@@ -0,0 +1,120 @@
+import { BoxProps } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { Autocomplete } from "@material-ui/lab";
+import React, { useEffect, useRef, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import Button from "../../../components/Button";
+import TextField from "../../../components/TextField";
+import { User } from "../../../pb/api_pb";
+import { service } from "../../../service";
+import { useAppDispatch, useTypedSelector } from "../../../store";
+import { fetchUsers, getUsers } from "../../userCache";
+
+const useStyles = makeStyles({ root: {} });
+
+export interface GroupChatListProps extends BoxProps {
+  createGroupChat: (title: string, users: User.AsObject[]) => void;
+}
+
+export function useFindUsers(ids: number[]): User.AsObject[] {
+  const notFound: number[] = [];
+  const usersRef = useRef<User.AsObject[]>([]);
+  const [idsString, setIdsString] = useState<string>("");
+  const allUsers = useTypedSelector((state) => getUsers(state));
+  const dispatch = useAppDispatch();
+  usersRef.current = [];
+  const users = usersRef.current;
+  for (const id of ids) {
+    const user = allUsers[id];
+    if (user) {
```

Comment:
From looking at the `fetchUsers` thunk, I believe it does this kind of checks there already and only fetch new users that don't already exist in the store. @lucaslcode will probably have more context on this.

If that's the case, you'll be able to simplify this a lot by just passing the whole list of `ids` you got from the prop to `fetchUsers`?